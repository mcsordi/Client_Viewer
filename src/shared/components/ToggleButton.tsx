import React from 'react';

export const ToggleButton = () => {
  return (
    <button className="cursor-pointer rounded-md h-10 px-6 bg-white flex md:hidden items-center">
      Mostrar
    </button>
  );
};
