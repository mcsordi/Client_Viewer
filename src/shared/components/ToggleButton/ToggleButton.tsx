import React, { useContext } from 'react';
import { toggle } from '../../../contexts/DrawerToggle/context.ts';
import { RiMenuFold4Fill } from 'react-icons/ri';

type TToggleButton = {
  visible?: boolean;
};
export const ToggleButton: React.FC<TToggleButton> = ({ visible }) => {
  const { show, toggleState } = useContext(toggle);
  return (
    <button
      className="absolute top-0 cursor-pointer rounded-md h-10 px-2 font-medium text-3xl
       bg-zinc-50 border border-zinc-400 flex sm:hidden items-center"
      onClick={() => toggleState()}
    >
      {show || visible ? (
        <RiMenuFold4Fill className="rotate-180 transition-all" />
      ) : (
        <RiMenuFold4Fill className="transition-all" />
      )}
    </button>
  );
};
