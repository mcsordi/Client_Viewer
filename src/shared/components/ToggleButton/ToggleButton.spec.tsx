import { render, screen } from '@testing-library/react';
import React from 'react';
import { ToggleButton } from './ToggleButton';

describe('<ToggleButton/>', () => {
  it('should render a button with a svg icon inside with class rotate-180 and transition-all', () => {
    const { container } = render(<ToggleButton visible={true} />);
    expect(container.firstChild?.firstChild).toHaveClass(
      'rotate-180 transition-all',
    );
  });
  it('should render a button with sgv icon inside', () => {
    const { container } = render(<ToggleButton visible={false} />);
    expect(container.firstChild?.firstChild).not.toHaveClass('rotate-180');
    expect(container.firstChild?.firstChild).toHaveClass('transition-all');
  });
  it('should match a snapshot', () => {
    const { container } = render(<ToggleButton visible={false} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
