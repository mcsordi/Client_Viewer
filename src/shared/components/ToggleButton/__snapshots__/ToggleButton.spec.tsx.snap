// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToggleButton/> should match a snapshot 1`] = `
<button
  class="absolute top-0 cursor-pointer rounded-md h-10 px-2 font-medium text-3xl bg-zinc-50 border border-zinc-400 flex sm:hidden items-center"
>
  <svg
    class="transition-all"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    stroke-width="0"
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M21 4H7V6H21V4ZM21 11H11V13H21V11ZM21 18H7V20H21V18ZM3 17V7L8 11.9996L3 17Z"
    />
  </svg>
</button>
`;
